### PR TITLE
Improve Command Add API in IM Command

### PR DESCRIPTION
**Pull request by yunhanw-google**
Summary of Changes:
-- Add PrepareCommand, GetCommandDataElementTLVWriter and FinishCommand
API to handle command add functionality, and remove old AddCommand which
create the redundant buffer to hold command data
-- Temporarily disable IM in experimental build since these API are used in ember, and we need to update ember in correspondence in separate PR and avoid large PR.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -58,7 +58,7 @@ jobs:
                      "gcc_release") GN_ARGS='is_debug=false';;
                      "clang") GN_ARGS='is_clang=true';;
                      "mbedtls") GN_ARGS='chip_crypto="mbedtls"';;
-                     "clang_experimental") GN_ARGS='is_clang=true chip_enable_interaction_model=true';;
```
Comment (msandstedt) [inline .github/workflows/build.yaml:61]: Is this change intentional?

Comment (yunhanw-google) [reply .github/workflows/build.yaml:61]: Yes -- Temporarily disable IM in experimental build since these API are used in ember IM, and we need to update ember in correspondence in separate PR and avoid large PR, the ticket is in https://github.com/project-chip/connectedhomeip/issues/6021

**Review by msandstedt — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Comment by woody-apple** [comment]
@pan-apple ?

**Review by yufengwangca — APPROVED** [review]

**Review by mspang — APPROVED** [review]
